### PR TITLE
:bug: Use visibility property instead of display to hide or show navbar

### PR DESCRIPTION
**Pull request by sandrohp88**
https://trello.com/c/zBPlQlcj

### CODE REVIEW

File: src/api/userApi.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { handleError, handleResponse } from './apiHelpers';
+
+// we should use .env variables for constants and keys
+const url = 'http://localhost:3002/friends/';
```
Comment (ArnasDickus) [inline src/api/userApi.js]: Due to plans, launching mappypals on Heroku. This url is not ideal. You should add something along with this.
` const url = process.env.URL || 'http://localhost:3001/';`

File: src/components/Navbar/Navbar.js [hunk 2]
```diff
@@ -9,6 +9,8 @@ import { ReactComponent as Friends } from '../../pics/FriendsIcon.svg';
 import { ReactComponent as Invite } from '../../pics/InviteFriends.svg';
 import { ReactComponent as Settings } from '../../pics/MySettingsIcon.svg';
 
+const VISIBILITY = { hidden: 'hidden', visible: 'visible' };
```
Comment (ArnasDickus) [inline src/components/Navbar/Navbar.js:12]: Could you explain What's the main idea with const VISIBILITY?

File: src/api/userApi.js (hunk 1 above)
Comment (sandrohp88) [reply src/api/userApi.js]: Totally agree!

File: src/store/actions/types.js [hunk 3]
```diff
@@ -1,3 +1,13 @@
...
+
+// user related actions
+export const USER_GET_FRIENDS_SUCCESS = 'USER_GET_FRIENDS_SUCCESS';
+
+// api status
+export const BEGIN_API_CALL = 'BEGIN_API_CALL';
+export const API_CALL_ERROR = 'API_CALL_ERROR';
+
+// api calls
+export const LOAD_USER_FRIENDS_API_CALL = 'LOAD_USER_FRIENDS_API_CALL';
```
Comment (ArnasDickus) [inline src/store/actions/types.js]: This needs to have a new line here. Try running npm run format

File: src/pages/FriendsList/FriendsList.js [hunk 4]
```diff
@@ -90,4 +103,15 @@ class FriendsList extends React.Component {
...
-export default FriendsList;
+// connect component to the store
+const mapStateToProps = ({ user, apiCallInProgress }) => ({
+    friends: user.friends,
+    loading: apiCallInProgress
+});
+const mapDispatchToProps = {
+    getUserFriends
+};
+export default connect(
```
Comment (ArnasDickus) [inline src/pages/FriendsList/FriendsList.js:114]: Due to connect there is a warning. To remove the warning in components >main.js. Change route to match login route.

warning

Change the path to similiar how it's in Login.

File: src/store/reduces/index.js [hunk 5]
```diff
@@ -1,9 +0,0 @@
-import { combineReducers } from 'redux';
```
Comment (ArnasDickus) [inline src/store/reduces/index.js:1]: Why did you delete this file?

File: src/api/userApi.js (hunk 1 above)
Comment (sandrohp88) [reply src/api/userApi.js]: I was wondering if can do this:
- create a file .env.development
- define REACT_APP_API_URL='http://localhost:3002/'
and then in the code
```javascript
const url = process.env.REACT_APP_API_URL + 'friends'
```

**Review by ArnasDickus — CHANGES_REQUESTED** [review]
Hello, thanks for adding PR. Your task was to fix a small bug. But it seems you added almost 1k lines of code. I'm going to need some explanations on what exactly did you do here.

On mobile I can't close the navbar. Clicking on the hamburger icon does nothing.

So this is what I gathered:
1) In navbar.js you added const visibility, which not sure what it does.
2) Added Friendliest better fake database. However, the original fake database still exists. In pages>friendList>friends-data.js

You should really avoid, working on multiple things, in one PR. It's harder to review. Also add instructions and guide on what did you changed.

I see most of the code is for the fake friendlist database. Including redux calls.

File: src/api/userApi.js (hunk 1 above)
Comment (ArnasDickus) [reply src/api/userApi.js]: I didn't realize that I added this as a single comment instead of an as a whole review. Anyways full review is now live @sandrohp88 Yes you can do that. There is a task for that. But you can implement it in this PR. https://trello.com/c/9gHrc6vx/111-adding-env-file-to-front-end

File: src/store/reduces/index.js (hunk 5 above)
Comment (sandrohp88) [reply src/store/reduces/index.js:1]: I just renamed the folder reduces to reducers

File: src/components/Navbar/Navbar.js (hunk 2 above)
Comment (sandrohp88) [reply src/components/Navbar/Navbar.js:12]: The idea is to favor the use of constants instead strings, for example:
```javascript
 { navbar: VISIBILITY.hidden }
 // instead of
  { navbar: 'hidden' }
```

Comment (sandrohp88) [reply src/components/Navbar/Navbar.js:12]: I tried to create a mock api with json-server and faker for fast prototyping (It should be created in a different PR but I messed it up). This way we could create fake data with few lines of code and the behavior will be more realistic (async). To add a new collection of data we just need to modify src/server/createMockDB.js 
```javascript
// declare collections here
const friends = [];

// generate fake data inside the loop
for (let i = 0; i < times; i++) {
    // generate fake friends
    friends.push({
        id: i,
        avatar: faker.internet.avatar(),
        name: faker.name.findName()
    });
}
// Add collectiont to fake db
const data = JSON.stringify({ friends });
```
This data will be added to src/server/db.json
You can now make fetch calls like:
```javascript
  const url = 'http://localhost:3002/collection/'
  const response = await fetch(url);
```
## Useful Links
+ [Faker](http://marak.github.io/faker.js/)
+ [json-server](https://github.com/typicode/json-server)

**Comment by mubarakshow** [comment]
@ArnasDickus Great review. 
So what's the state of this PR, @sandrohp88 seems to have explained himself on your comments. So will this be merged or redone?

**Review by ArnasDickus — CHANGES_REQUESTED** [review]
@sandrohp88  I see you fixed almost all the mistakes.

I still have a few concerns:
 1) Since you added a better way for fake DB. You need to delete the old fake database, friends-data.js, and possibly others not used files related to friendlist.
 2) Looking at friendlist [mockup](https://trello.com/c/v4TFipnX/109-friendslist). There should be 4 cards, not 6 displayed in a single area.
3) Regarding `const url = process.env.URL || 'http://localhost:3001/';`. You can fix this issue with new PR or add this into this PR. I would approve it regardless.

**Comment by sandrohp88** [comment]
1. I forgot to remove friends-data.js (sorry about that)
2. I don't remember changing  friendlist's styles but I looked into it and cards wrap to fit the screen size.
3. ```javascript const url = process.env.URL || 'http://localhost:3001/';```
 Used this approach now, probably we will change it in a future PR

**Review by ArnasDickus — APPROVED** [review]

**Comment by ArnasDickus** [comment]
Alright, I see you fixed all the changes. Except for displaying 6 friends. We could fix that later.
